etl/snapshot.py

```diff
@@ -353,7 +371,7 @@ def to_table_metadata(self):
 def add_snapshot(
     uri: str,
     filename: Optional[Union[str, Path]] = None,
-    dataframe: Optional[pd.DataFrame] = None,
+    dataframe: Optional[Union[Table, pd.DataFrame]] = None,
```

Comment:
In case you are interested, you can use the following syntax now (since [python 3.10](https://peps.python.org/pep-0604/)):

```python
dataframe: Optional[Table | pd.DataFrame]
```